pkg/meta/sql.go

```diff
@@ -3900,3 +3907,37 @@ func (m *dbMeta) doAttachDirNode(ctx Context, parent Ino, inode Ino, name string
 		return err
 	}, parent))
 }
+
+func (m *dbMeta) touchAtime(ctx Context, ino Ino) syscall.Errno {
```

Comment:
We'd better make the decision of "shall we update the atime" based on different operations (open, readdir, read, etc.), not in a general function. This is because metadata transactions are expensive, and should be avoided whenever possible.

Reply:
In other words, this function can be used to UPDATE the atime, while the CHECK should be done before calling it.

Reply:
OK, I updated touchAtime() and avoided transaction if possible, but still in check & update manner.